src/codegen/translate-expression.lisp

```diff
@@ -550,6 +551,139 @@ Returns a `node'.")
                           :type tc:*unit-type*
                           :value unit-value))))))
 
+  (:method ((expr tc:node-while) ctx env)
+    (declare (type pred-context ctx)
+             (type tc:environment env)
+             (values node))
+    
+    (make-node-while 
+     :type tc:*unit-type*
+     :label (tc:node-while-label expr)
+     :expr (translate-expression (tc:node-while-expr expr) ctx env)
+     :body (translate-expression (tc:node-while-body expr) ctx env)))
+
+  (:method ((expr tc:node-while-let) ctx env)
+    (declare (type pred-context ctx)
+             (type tc:environment env)
+             (values node))
+    (make-node-while-let
+     :type tc:*unit-type*
+     :pattern (translate-pattern (tc:node-while-let-pattern expr))
+     :label (tc:node-while-let-label expr)
+     :expr (translate-expression (tc:node-while-let-expr expr) ctx env)
+     :body (translate-expression (tc:node-while-let-body expr) ctx env)))
+
+  (:method ((expr tc:node-for) ctx env)
```

Comment:
Can `tc:node-for` translate into an `ast:node-while-let`?

Reply:
this is not very convenient